heartbeat/autodiscover/builder/hints/heartbeats_test.go

```diff
@@ -0,0 +1,216 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package hints
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/logp"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/bus"
+)
+
+func TestGenerateHints(t *testing.T) {
+	tests := []struct {
+		message string
+		event   bus.Event
+		len     int
+		result  common.MapStr
+	}{
+		{
+			message: "Empty event hints should return empty config",
+			event: bus.Event{
+				"host": "1.2.3.4",
+				"kubernetes": common.MapStr{
+					"container": common.MapStr{
+						"name": "foobar",
+						"id":   "abc",
+					},
+				},
+				"docker": common.MapStr{
+					"container": common.MapStr{
+						"name": "foobar",
+						"id":   "abc",
+					},
+				},
+			},
+			len:    0,
+			result: common.MapStr{},
+		},
+		{
+			message: "Hints without host should return nothing",
```

Comment:
Going back to earlier comments, if there are some heartbeat hints required, but not others, we should log an error. It'd be great to check for that log message in our tests. It might be a bit challenging with our current code structure. One thought is that the builder could return warnings (maybe just as a `[]string`) in addition to errors. That'd be easy to test.